core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -401,6 +402,16 @@ private static int makeRegexpFlags(@Nullable String stringFlags) {
     return flags;
   }
 
+  /** SQL {@code REGEXP_CONTAINS(value, regexp)} function. */
+  public static boolean regexpContains(String value, String regex) {
+    try {
+      Pattern regexp = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
```

Comment:
Does this pattern handle Java regular expressions or BigQuery regular expressions?

Reply:
Hi @mihaibudiu, this Pattern object is an instance of the java.util.regex library, it has almost complete overlap with BQ regex except for a few negative scenarios as highlighted on this ticket: [CALCITE-5910](https://issues.apache.org/jira/browse/CALCITE-5910)